hack/verify-release.sh

```diff
@@ -0,0 +1,574 @@
+    fi
+
+    repo_status="$(git diff --stat)"
+    if [[ -n "${repo_status}" ]]; then
+        echo "WARNING: your local repository is dirty"
+    fi
+
+    echo -e "Done\n"
+}
+
+download_release_information()
+{
+    # download release information json, requires GITHUB_TOKEN
+    echo "Downloading release information ..."
+
+    if ! curl -SsL --fail \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${GITHUB_TOKEN}" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            -o "${RELEASE_JSON}" \
+            "https://api.github.com/repos/${PROJECT}/releases/tags/v${VERSION}" >/dev/null; then
+        echo "ERROR: could not download release information, check token and permissions"
+        exit 1
+    fi
+
+    echo -e "Done\n"
+}
+
+
+#
+# verification functions
+#
+verify_git_tags()
+{
+    # check tags exist in remote, ie. are not just local but pushed
+    echo "Verifying Git tags ..."
+
+    for tag in "${git_annotated_tags[@]}" "${git_lightweight_tags[@]}"; do
+        if ! git ls-remote --exit-code --tags "${REMOTE}" "refs/tags/v${VERSION}" &>/dev/null; then
+            echo "ERROR: tag ${tag} is not found in remote ${REMOTE}"
+        fi
+    done
+
+    echo -e "Done\n"
+}
+
+verify_git_tag_types()
+{
+    # check tags are annotated or lightweight as expected
+    # and also that no extra tags are pushed by accident
+    echo "Verifying Git tag types ..."
+
+    for annotated_tag in "${git_annotated_tags[@]}"; do
+        if [[ "$(git cat-file -t "${annotated_tag}" 2>/dev/null)" != "tag" ]]; then
+            echo "ERROR: ${annotated_tag} is not an annotated tag, or is missing"
+        fi
+    done
+
+    for lightweight_tag in "${git_lightweight_tags[@]}"; do
+        if [[ "$(git cat-file -t "${lightweight_tag}" 2>/dev/null)" != "commit" ]]; then
+            echo "WARNING: ${lightweight_tag} is not a lightweight tag, or is missing"
+        fi
+    done
+
+    for nonexist_tag in "${git_nonexisting_tags[@]}"; do
+        if git cat-file -t "${nonexist_tag}" &>/dev/null; then
+            echo "ERROR: ${nonexist_tag} is exists, while it should not"
+        fi
+    done
+
+    echo -e "Done\n"
+}
+
+verify_release_notes()
+{
+    # check release note content
+    echo "Verifying release notes ..."
+
+    # check body if certain strings
+    for string in "${release_note_tag_strings[@]}"; do
+        # shellcheck disable=SC2076
+        if ! [[ "$(jq .body "${RELEASE_JSON}")" =~ "${string}" ]]; then
+            echo "ERROR: '${string}' not found in release note text, is tag correct?"
+        fi
+    done
+
+    # check body for tagged images
+    for string in "${release_note_strings[@]}"; do
+        # shellcheck disable=SC2076
+        if ! [[ "$(jq .body "${RELEASE_JSON}")" =~ "${string}" ]]; then
+            echo "WARNING: '${string}' not found in release note text, recheck content"
+        fi
+    done
+
+    echo -e "Done\n"
+}
+
+verify_release_artefacts()
```

Comment:
artifacts

Reply:
Quite a few of this in the file, you can fix all

Reply:
Artefact is British English spelling, artifact is just Americans mangling proper English :)

Reply:
Ah, true. I am more used to artifact , in my knowledge base artefact never existed and just seemed wrong 😄